tracepoints/operational/runtest.sh

```diff
@@ -0,0 +1,232 @@
+    local PROBES_FILE=`mktemp -p /tmp -t group.XXXXXX`
+    local PROBES_NAME_FILE=`mktemp -p /tmp -t group_probe_names.XXXXXX`
+    for i in $TESTLIST; do
+
+        # Create unique log for verbose tracepoint logging
+        COUNT=`expr $COUNT + 1`
+        STIME=`date +%s`
+
+        echo "$i" | grep -q "hcall_"
+        if [ $? -eq 0 ]; then
+                echo "Skipping probe $i due to Bug 1143870" | tee -a $OUTPUTFILE
+                echo "Bug 1143870 - ppc64le kernel hangs while running systemtap with hcall_entry/hcall_exit probes" | tee -a $OUTPUTFILE
+                continue
+        fi
+
+        echo "$i" | tr -d '\"' >> $PROBES_NAME_FILE
+        echo 'probe kernel.trace('$i') { if (pid() == 0) printf("probe hit\n"); }' >> $PROBES_FILE
+
+        if [ $((COUNT % GROUP_SIZE)) == 0 -o $COUNT == $PROBE_COUNT ]; then
+
+            testHeader
+            echo "------------------------------------------------------------" | tee -a $OUTPUTFILE
+            echo "         Start of SystemTap Kernel Tracepoint Test          " | tee -a $OUTPUTFILE
+            echo "         $STIME                                             " | tee -a $OUTPUTFILE
+            echo "         Testing:                                           " | tee -a $OUTPUTFILE
+            cat $PROBES_FILE | tee -a $OUTPUTFILE
+            echo "------------------------------------------------------------" | tee -a $OUTPUTFILE
+
+            cat $PROBES_FILE > group.stap
+
+            local firstp=`head -1 $PROBES_NAME_FILE`
+            local lastp=`tail -1 $PROBES_NAME_FILE`
+            local VAR="$firstp"
+            if [ ! "$firstp" == "$lastp" ]; then
+                local VAR="${firstp}__to__${lastp}"
+            fi
+            local VERBOSETRACELOG=`mktemp -p /mnt/testarea -t $VAR-TraceLog.XXXXXX`
+
+            stap -DSTP_NO_OVERLOAD $XTRA -t -c "sleep 0.25" ${STAP_VERBOSE_FLAG} group.stap > $VERBOSETRACELOG 2>&1
+            local rc=$?
+            timeCalc
+            if [ $rc -eq 0 ] ; then
+                echo "       Result testing : Test Passed " | tee -a $OUTPUTFILE
+                echo "       Test run time  : $TTIME seconds " | tee -a $OUTPUTFILE
+                echo "------------------------------------------------------------" | tee -a $OUTPUTFILE
+                resultPass $VAR $COUNT
+            else
+                echo "       Result testing : Test Failed " | tee -a $OUTPUTFILE
+                echo "       Test run time  : $TTIME seconds " | tee -a $OUTPUTFILE
+                echo "------------------------------------------------------------" | tee -a $OUTPUTFILE
+                submitLog $VERBOSETRACELOG
+                resultFail $VAR $COUNT
+            fi
+            rm -f $PROBES_FILE
+            rm -f $PROBES_NAME_FILE
+        fi
+    done
+}
+
+function runTest ()
+{
+    local TESTLIST=`/usr/bin/stap -L 'kernel.trace("*")' | grep -o "\".*\""`
+    if [ -z "$TESTLIST" ] ; then
+        resultFail TESTLIST_EMPTY 99
```

Comment:
I'm not sure if this is something we should report as a failure or a prerequisite for the testing (in which case we should skip or abort), as I'm not familiar with this at all. @ftjahjad can you check which one it is?

Reply:
If SystemTap cannot find any kernel tracepoints, I would consider that a failure.  Unless you intend to build a kernel with tracepoints disabled, this should be a failure (not an abort).

Reply:
@jbastian , thanks. it is clear now.

Reply:
We either build normal tarballs (upstream) or use rpmbuild with the following options:

```--with=kabidw_base --without=bpftool --without=debug --without=doc --without=ipaclones --without=perf --without=selftests --without=tools
```
The configs used are the standard configs used with the release.
So I think the building part should include tracepoints and everything should be ok. Thanks @jbastian !